app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/SitePageActivity.java

```diff
@@ -45,6 +53,15 @@ protected void onCreate(Bundle savedInstanceState) {
             startSiteAnnouncementsFragment(course);
         } else if (siteType.equals(getString(R.string.resources_site))) {
             startSiteResourcesFragment(course);
+        } else if (siteType.equals(getString(R.string.assignments_site))) {
+            if(course != null)
```

Comment:
The `Intent` to open `SitePageActivity` from the assignments tab doesn't come from the `AssignmentsFragment` itself, but rather from the `AssignmentAdapter`, which is not aware of courses (the adapter only contains a list of assignments as it doesn't need to worry about the respective courses). Also, this wouldn't really make a difference because we want to listen from the database anyways, so showing data passed in the through the `Bundle` would be an anti-pattern relative to what we've been aiming for and working towards.